zarr/core.py

```diff
@@ -155,6 +161,7 @@ def __init__(
         cache_attrs=True,
         partial_decompress=False,
         write_empty_chunks=True,
+        meta_array=None,
```

Comment:
I wonder if this could be picked up from the `Store` itself. That way a `Store` could specify its return type (NumPy or otherwise). This would save the user from getting involved as much in the process and hopefully make it easier for them to get started. WDYT?

@grlee77 would be good to get your thoughts as well in light of the `BaseStore` work 🙂

Reply:
I think it would be a good idea to have `Store` define a default `meta_array` but in the general case, I think a `Store` should be able to handle both NumPy and other types like CuPy arrays simultaneously.